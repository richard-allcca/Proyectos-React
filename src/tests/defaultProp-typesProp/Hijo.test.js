import { shallow } from "enzyme";
import Hijo from "./../../conceptos_basicos/defaultProp-typesProp/Hijo";

describe("Prueba del componente Hijo", () => {
  test("Retorna el contenido html del hijo", () => {
    const wrapper = shallow(<Hijo saludo="hola desde el hijo Richard" />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Retorna el subtitulo enviado por props", () => {
    const saludo = "test para obtener elementos del componente";
    const numero = 2;
    const subtitle = "Soy el subTitulo";

    const wrapper = shallow(
      <Hijo numero={numero} saludo={saludo} subtitle={subtitle} />
    );

    const textoParrafo = wrapper.find("p").at(0).text();

    expect(textoParrafo).toBe(saludo);
  });
});

// Notas:
// mas detalles en las notas en el "Contador.test.js"
// 1. find() actua como d.querySelector('h2) apunta a un elemento html')
// 2. text() extrae el texto del elemento seleccionado con find()
