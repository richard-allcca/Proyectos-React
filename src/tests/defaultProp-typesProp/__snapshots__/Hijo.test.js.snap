// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente Hijo Retorna el contenido html del hijo 1`] = `
<div>
  <h2>
    12
  </h2>
  <p>
    hola desde el hijo Richard
  </p>
  <h3>
    Soy un subTitulo
  </h3>
</div>
`;
